src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/DSDropDownBase.cs

```diff
@@ -15,61 +15,71 @@ namespace CoreNodeModelsWpf.Nodes
         {
             this.model = model;
 
-            //add a drop down list to the window
-            var combo = new ComboBox
+            // Add a drop down list to the window
+            comboBox = new ComboBox
             {
-                Width = System.Double.NaN,
-                MinWidth= 150,
+                Width = double.NaN,
+                MinWidth = 150,
                 Height = Configurations.PortHeightInPixels,
                 HorizontalAlignment = HorizontalAlignment.Stretch,
-                VerticalAlignment = VerticalAlignment.Center
+                VerticalAlignment = VerticalAlignment.Center,
+                Style = (Style)SharedDictionaryManager.DynamoModernDictionary["RefreshComboBox"]
             };
 
+            nodeView.inputGrid.Children.Add(comboBox);
+            Grid.SetColumn(comboBox, 0);
+            Grid.SetRow(comboBox, 0);
 
-            combo.Style = (Style)SharedDictionaryManager.DynamoModernDictionary["RefreshComboBox"];
+            comboBox.DropDownOpened += DropDownOpened;
+            comboBox.SelectionChanged += SelectionChanged;
 
-            nodeView.inputGrid.Children.Add(combo);
-            System.Windows.Controls.Grid.SetColumn(combo, 0);
-            System.Windows.Controls.Grid.SetRow(combo, 0);
+            comboBox.DataContext = model;
 
-            combo.DropDownOpened += combo_DropDownOpened;
-
-            combo.DataContext = model;
-
-            // bind this combo box to the selected item hash
-            var bindingVal = new System.Windows.Data.Binding("Items")
+            // Bind this combo box to the selected item hash.
+            var bindingVal = new Binding(nameof(DSDropDownBase.Items))
             {
                 Mode = BindingMode.TwoWay,
                 Source = model
             };
-            combo.SetBinding(ItemsControl.ItemsSourceProperty, bindingVal);
+            comboBox.SetBinding(ItemsControl.ItemsSourceProperty, bindingVal);
             
-            // bind the selected index to the model property SelectedIndex
-            var indexBinding = new Binding("SelectedIndex")
+            // Bind the selected index to the model property SelectedIndex.
+            var indexBinding = new Binding(nameof(DSDropDownBase.SelectedIndex))
             {
                 Mode = BindingMode.TwoWay,
                 Source = model
             };
-            combo.SetBinding(Selector.SelectedIndexProperty, indexBinding);
+            comboBox.SetBinding(Selector.SelectedIndexProperty, indexBinding);
+        }
+
+        private void SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            if (comboBox.SelectedIndex != -1)
+            {
+                model.OnNodeModified();
+            }
         }
 
         public void Dispose()
         {
+            comboBox.DropDownOpened -= DropDownOpened;
+            comboBox.SelectionChanged -= SelectionChanged;
+
+            GC.SuppressFinalize(this);
```

Comment:
I know reshaper might suggest this, but because we're kind of mis-using IDisposable, and not dealing with any unmanaged resources I would leave out the suppress finalizer call unless you have a specific reason.